File: autonoma1semestre2.py
# Definimos funciones para manejar el repositorio

def mostrar_libros(repositorio):
    print(" Lista de Libros ")
    for libro in repositorio:
        print(f"Título: {libro['titulo']} | Autor: {libro['autor']} | Anio: {libro['anio']}")

def eliminar_libro(repositorio, titulo):
    encontrado = False
    for libro in repositorio:
        if libro['titulo'] == titulo:
            repositorio.remove(libro)
            print("Libro eliminado correctamente.")
            encontrado = True
            break
    if not encontrado:
        print("No se encontró el libro con ese título.")

def ordenar_por_anio(repositorio):
    repositorio.sort(key=lambda l: l['anio'])
    print("Libros ordenados por anio.")

# Programa principal
repositorio = []

# Ingreso de al menos 3 libros
for i in range(3):
    titulo = input("Ingrese titulo del libro: ")
    autor = input("Ingrese autor del libro: ")
    anio = int(input("Ingrese anio de publicacion: "))
    repositorio.append({"titulo": titulo, "autor": autor, "anio": anio})

mostrar_libros(repositorio)

# Agregar un libro adicional
print("Agregar un nuevo libro")
titulo = input("Titulo: ")
autor = input("Autor: ")
anio = int(input("Anio: "))
repositorio.append({"titulo": titulo, "autor": autor, "anio": anio})

mostrar_libros(repositorio)

# Eliminar un libro por título
titulo_eliminar = input("Ingrese el título del libro a eliminar: ")
eliminar_libro(repositorio, titulo_eliminar)

mostrar_libros(repositorio)

# Ordenar por año
ordenar_por_anio(repositorio)
mostrar_libros(repositorio)

File: autonoma1semestre2.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

// Definimos la estructura Libro
struct Libro {
    string titulo;
    string autor;
    int anio;
};

// Función para mostrar los libros
void mostrarLibros(const vector<Libro>& repositorio) {
    cout << " Lista de Libros ";
    for (const auto& libro : repositorio) {
        cout << "Titulo: " << libro.titulo
             << " Autor: " << libro.autor
             << "  Año: " << libro.anio << endl;
    }
}

// Función para eliminar un libro por título
void eliminarLibro(vector<Libro>& repositorio, const string& titulo) {
    auto it = remove_if(repositorio.begin(), repositorio.end(),
                        [&](const Libro& l){ return l.titulo == titulo; });
    if (it != repositorio.end()) {
        repositorio.erase(it, repositorio.end());
        cout << "Libro eliminado correctamente.";
    } else {
        cout << "No se encontró el libro con ese titulo.";
    }
}

// Función para ordenar por año
void ordenarPorAnio(vector<Libro>& repositorio) {
    sort(repositorio.begin(), repositorio.end(),
         [](const Libro& a, const Libro& b){ return a.anio < b.anio; });
    cout << "Libros ordenados por año.";
}

int main() {
    vector<Libro> repositorio;

    // Ingreso de al menos 3 libros
    for (int i = 0; i < 3; i++) {
        Libro libro;
        cout << "Ingrese titulo del libro: ";
        getline(cin, libro.titulo);
        cout << "Ingrese autor del libro: ";
        getline(cin, libro.autor);
        cout << "Ingrese anio de publicacion: ";
        cin >> libro.anio;
        cin.ignore(); // limpiar buffer
        repositorio.push_back(libro);
    }

    mostrarLibros(repositorio);

    // Agregar un libro adicional
    Libro nuevo;
    cout << "Agregar un nuevo libro ";
    cout << "Titulo: ";
    getline(cin, nuevo.titulo);
    cout << "Autor: ";
    getline(cin, nuevo.autor);
    cout << "Anio: ";
    cin >> nuevo.anio;
    repositorio.push_back(nuevo);

    mostrarLibros(repositorio);

    // Eliminar un libro por título
    string tituloEliminar;
    cout << "Ingrese el titulo del libro a eliminar: ";
    cin.ignore();
    getline(cin, tituloEliminar);
    eliminarLibro(repositorio, tituloEliminar);

    mostrarLibros(repositorio);

    // Ordenar por año
    ordenarPorAnio(repositorio);
    mostrarLibros(repositorio);

    return 0;
}
